modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/side_navigation.es.js

```diff
@@ -427,6 +447,206 @@ SideNavigation.prototype = {
 		}
 	},
 
+	_onClosed() {
+		const instance = this;
+		const options = instance.options;
+
+		const container = document.querySelector(options.container);
+
+		if (!instance._handleClosed) {
+			instance._handleClosed = () => {
+				const type = !instance._isDesktop()
+					? options.typeMobile
+					: options.type;
+
+				if (type === 'relative' && hasClass(container, 'open')) {
+					removeClass(container, 'sidenav-transition');
+				}
+			};
+
+			document.addEventListener(
+				'closed.lexicon.sidenav',
+				instance._handleClosed
+			);
+		}
+	},
+
+	_onClosedStart() {
+		const instance = this;
+		const options = instance.options;
+
+		const container = document.querySelector(options.container);
+		const content = document.querySelector(options.content);
+
+		if (!instance._handleClosedStart) {
+			instance._handleClosedStart = () => {
+				const type = !instance._isDesktop()
+					? options.typeMobile
+					: options.type;
+
+				if (
+					type === 'relative' &&
+					hasClass(container, 'open') &&
+					content.closest('.page-maximized')
+				) {
+					let contentMargin =
+						document.body.scrollWidth -
+						content.getBoundingClientRect().right;
+
+					let paddingRight = options.gutter + options.width;
+
+					const contentMaxWidth =
+						getComputedStyle(content).maxWidth ||
+						getComputedStyle(content).width;
+
+					if (contentMaxWidth.match(/px$/g)) {
+						contentMargin =
+							(document.body.scrollWidth -
+								toInt(contentMaxWidth)) /
+							2;
+
+						if (contentMargin > options.width) {
+							paddingRight = '';
+						}
+						else if (
+							contentMargin > 0 &&
+							contentMargin < options.width
+						) {
+							paddingRight =
+								options.gutter + options.width - contentMargin;
+						}
+					}
+
+					addClass(container, 'sidenav-transition');
+
+					setStyles(content, {
+						'padding-right': px(paddingRight),
+					});
+				}
+			};
+
+			document.addEventListener(
+				'closedStart.lexicon.sidenav',
+				instance._handleClosedStart
+			);
+		}
+	},
+
+	_onOpen() {
+		const instance = this;
+		const options = instance.options;
+
+		const container = document.querySelector(options.container);
+
+		if (!instance._handleOpen) {
+			instance._handleOpen = () => {
+				const type = !instance._isDesktop()
+					? options.typeMobile
+					: options.type;
```

Comment:
Another ternary that would be more readable if inverted to avoid the negation. I see two, three, four(?) more instances below as well, giving us around six in total. At that point I'd just move this into a method and call it in those four places; ie:

```js
const type = this._getType();
```

with implementation:

```js
_getType() {
    return this._isDesktop() ? this.options.type : this.options.typeMobile;

    // or this, if you want it to be more shorter:
    //return this.options[this._isDesktop() ? 'type': 'typeMobile'];
}
```